pyglossary/plugins/ebook_kobo.py

```diff
@@ -162,14 +169,44 @@ def writeGroup(lastPrefix):
 				gzipFile.write(htmlContents.encode("utf-8"))
 			htmlContents = htmlHeader
 
-		lastPrefix = ""
+		data = []
+
 		while True:
 			entry = yield
 			if entry is None:
 				break
 			if entry.isData():
 				dataEntryCount += 1
 				continue
+			l_word = entry.l_word
+			if len(l_word) == 1:
+				data.append(entry.getRaw(glos))
+				continue
+			wordsByPrefix = OrderedDict()
+			for word in l_word:
+				prefix = self.get_prefix(word)
+				if prefix in wordsByPrefix:
+					wordsByPrefix[prefix].append(word)
+				else:
+					wordsByPrefix[prefix] = [word]
+			if len(wordsByPrefix) == 1:
+				data.append(entry.getRaw(glos))
+				continue
+			defi = entry.defi
+			for prefix, p_words in wordsByPrefix.items():
+				for word in l_word:
+					if word not in p_words:
+						p_words.append(word)
+				data.append(Entry(p_words, defi).getRaw(glos))
+			del entry
+
+		log.info(f"\nKobo: sorting entries...")
+		data.sort(key=Entry.getRawEntrySortKey(self.get_prefix_b))
+
+		lastPrefix = ""
+		for rawEntry in data:
+			entry = Entry.fromRaw(glos, rawEntry)
+
 			headword, *variants = entry.l_word
```

Comment:
Because the keywords in the newly synthesized entries don't have the original order, this does not yield the correct headword.

Reply:
I thought this is what we meant to do. What problem does this cause?

Reply:
If we have an entry "go" with the alternatives "went" and "gone", we (or at least I) want to show "go" as the title of the entry, even when the user searches for "went". Otherwise the entry will look irritating like

> **went**
> verb
> the act of moving by foot

IMO the entry should read

> **go**
> verb
> the act of moving by foot

Which could be improved in the future (needs some more work on my side) to

> to **go**
> verb
> The act of moving by foot
> Conjugations: went, gone

Reply:
Got it. I'm merging this. I will add a new PR for that later.